### ISSUE TITLE
Failed to load plugin html: Cannot find module 'eslint-plugin-html'

### ISSUE THREAD

**Issue by vanshady** [Microsoft/vscode-eslint#64]
I already installed almost everything and I could manually run eslint through shell, but when I was using vs code, this warning kept popping up.

**Comment by vanshady** [comment]
I installed eslint-plugin-html both globally and locally, but it still doesn't work

**Comment by Beingbook** [comment]
True. Now `vscode-eslint` is not able to find plugins such as `eslint-plugin-xxx` even if the dependencies are installed correctly.

**Comment by dbaeumer** [comment]
I couldn't reproduce the failing plugin loading. This works for me if the plugins are installed locally. However there is a special issue with the html plugin. It is designed to validate script tags html files however html files are not passed to the eslint server from VSCode. And they should only be passed if the html plugin is installed. Need to think about how to make this work nicely.

**Comment by vanshady** [comment]
The error popped-up when I was working on jsx. I installed the following as my devDependencies
```
    "eslint": "^2.4.0",
    "eslint-config-airbnb": "^6.1.0",
    "eslint-plugin-html": "^1.4.0",
    "eslint-plugin-react": "^4.3.0"
```
and here is my eslintrc
```
{
    "extends": "airbnb",
    "plugins": [
      "html"
    ],
    "rules": {
      "react/prefer-es6-class": [2, "never"],
      "react/prefer-stateless-function": 0,
      "no-console": 0
    }
}
```

**Comment by ChuckPierce** [comment]
@vanshady This is error is probably happening because you don't have your locally installed `node_modules` in the root of your project.  You've probably opened a project that doesn't have a JS backend and your node_modules folder is living in the second level of your project tree.

**Comment by chrisdhanaraj** [comment]
I have the same problem - and my node_modules folder is in the root of my project

**Comment by mamboer** [comment]
same issue

**Comment by dbaeumer** [comment]
@chrisdhanaraj @mamboer is one of your workspace public so that I could try to reproduce this. As said it works in my simple sample setup.

**Comment by mamboer** [comment]
@dbaeumer try this repo https://github.com/yuche/vue-strap

**Comment by dbaeumer** [comment]
@mamboer I cloned your git repository and run npm install (node 4.4.5). This neither installes eslint nor the eslint-html plugin nor babel-eslint. Installing all three globally makes the vscode-eslint extension work without problems:

![capture](https://cloud.githubusercontent.com/assets/1931590/16296427/8480df28-392c-11e6-98f5-53ba18ee0072.png)


Am I missing something?

**Comment by mamboer** [comment]
@dbaeumer My node version is 6.2.0, can you switch your node to this version and try one more?

**Comment by dbaeumer** [comment]
I see the same with node 6.2.0.

**Comment by chrisdhanaraj** [comment]
Hey @dbaeumer huge apologies for the delay on this. I pushed up a sample of how I expected eslint to work.

https://github.com/chrisdhanaraj/eslint-test

If you run `npm run lint`, it uses the local eslint to the package and works fine - VSCode complains that it cannot find the package though

**Comment by chrisdhanaraj** [comment]
@dbaeumer okay jokes, figured it out. My scenario

At one point early on, I had my global `node_modules` saved to $HOME/node_modules (I had since moved to a `nvm` controlled environment and completely spaced about it). VSCode was using that old `node_modules` folder and the eslint package in there. When I deleted those files, VSCode started using my local install again.

This is a little different than what the Atom/Sublime plugins do, and kind of what I expected VSCode to do.

**Comment by vanshady** [comment]
@chrisdhanaraj Thank you so much! Finally, there's a solution.

**Comment by yanlee26** [comment]
VS integrated with ESLint, this seems not needed now
